### PR TITLE
Updated youtube-dl-gui to v1.8.3

### LINKED ISSUE #7815: Error with yt-dlg
## Bug Report

**Package Name:** yt-dlg

### Current Behaviour

Does not install

### Additional context/output

```
Installing 'youtube-dl-gui' (1.8.2) [64bit]
Loading yt-dlg.msi from cache
Checking hash of yt-dlg.msi ... ok.
Running pre-install script...
Could not find 'yt-dlg'! (error 16)
At C:\Users\Novitzka\scoop\apps\scoop\current\lib\core.ps1:516 char:9
+         throw "Could not find '$(fname $from)'! (error $($proc.ExitCo ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Could not find 'yt-dlg'! (error 16):String) [], RuntimeException
    + FullyQualifiedErrorId : Could not find 'yt-dlg'! (error 16)
```

### Possible Solution

N/A

### System details

**Windows version:** 10

**OS architecture:** 64

**PowerShell version:** N/A

**Additional software:** N/A

#### Scoop Configuration

```json
{
    "lastupdate":  "2022-01-22T15:50:57.4645226+01:00",
    "SCOOP_REPO":  "https://github.com/ScoopInstaller/Scoop",
    "SCOOP_BRANCH":  "master"
}
```

### PR DESCRIPTION
**Pull request by brian6932**
Just updated the package, this version had a name change in the .msi and was not being auto updated, I am not entirely sure if I got the `checkver` and `autoupdate` syntax correct, so pls lmk

Closes #7815

### CODE REVIEW

File: bucket/youtube-dl-gui.json [hunk 1]
```diff
@@ -20,15 +20,21 @@
...
+        "github":"https://github.com/oleksis/youtube-dl-gui",
+        "re":"(?<date>(-[\\d\\.]+)?)"
+    },
     "autoupdate": {
         "architecture": {
             "64bit": {
-                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/yt-dlg.msi#/dl.ism"
+                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/yt-dlg$matchDate.msi#/dl.ism",
+                "hash":{
+                    "url": "$baseurl/releases/download/v$version/SHA2-256SUMS"
```
Comment (HUMORCE) [inline bucket/youtube-dl-gui.json:36]:
```suggestion
                    "url": "$baseurl/SHA2-256SUMS"
```

**Review by HUMORCE — CHANGES_REQUESTED** [review]

File: bucket/youtube-dl-gui.json [hunk 2]
```diff
@@ -20,15 +20,21 @@
     ],
     "shortcuts": [
         [
-            "yt-dlg.exe",
+            "yt-dlg\\yt-dlg.exe",
```
[the remark this answers is not in the document]
Comment (brian6932) [reply bucket/youtube-dl-gui.json:23]: @HUMORCE I tried this, but I was getting errors saying that the executable could not be found, but once I added `yt-dlg\\` it fixed the issue

**Comment by brian6932** [comment]
I get the same errors when my manifest looks like this:
```json
{
    "version": "1.8.3",
    "description": "A cross platform front-end GUI of the popular youtube-dl written in wxPython.",
    "homepage": "https://github.com/oleksis/youtube-dl-gui",
    "license": "Unlicense",
    "architecture": {
        "64bit": {
            "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v1.8.3/yt-dlg-20220118.2.msi#/dl.ism",
            "hash": "71d4fc4eea97199218fdde36717e90326ed0fd4bd980c6afbffc263514e34be9"
        }
    },
    "pre_install": [
        "$_args = @{",
        "    'Path' = \"$dir\\$fname\"",
        "    'DestinationPath' = \"$dir\"",
        "    'Removal' = $true",
        "}",
        "if (get_config 'MSIEXTRACT_USE_LESSMSI' $true) { $_args.Add('ExtractDir', 'APPDIR\\yt-dlg') } else { $_args.Add('ExtractDir', 'yt-dlg') }",
        "Expand-MsiArchive @_args"
    ],
    "shortcuts": [
        [
            "yt-dlg.exe",
            "yt-dlg"
        ]
    ],
    "checkver": {
        "github":"https://github.com/oleksis/youtube-dl-gui",
        "regex": "v(?<version>[\\d.]+)/(?<msi>.*).msi"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/$matchMsi.msi#/dl.ism",
                "hash":{
                    "url": "$baseurl/SHA2-256SUMS"
                }
            }
        }
    }
}
```
Console log:
```pwsh
❯ scoop install .\youtube-dl-gui.json
WARN  Purging previous failed installation of youtube-dl-gui.
Uninstalling 'youtube-dl-gui' ().
'youtube-dl-gui' was uninstalled.
Installing 'youtube-dl-gui' (1.8.3) [64bit]
Loading yt-dlg-20220118.2.msi from cache
Checking hash of yt-dlg-20220118.2.msi ... ok.
Running pre-install script...
Exception: C:\Users\username\scoop\apps\scoop\current\lib\core.ps1:516
Line |
 516 |          throw "Could not find '$(fname $from)'! (error $($proc.ExitCo …
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Could not find 'yt-dlg'! (error 16)
```
However, I get no errors, when it doesn't use `#/dl.ism` and `pre_install`, and just install the .msi:
```json
{
    "version": "1.8.3",
    "description": "A cross platform front-end GUI of the popular youtube-dl written in wxPython.",
    "homepage": "https://github.com/oleksis/youtube-dl-gui",
    "license": "Unlicense",
    "architecture": {
        "64bit": {
            "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v1.8.3/yt-dlg-20220118.2.msi",
            "hash": "71d4fc4eea97199218fdde36717e90326ed0fd4bd980c6afbffc263514e34be9"
        }
    },
    "shortcuts": [
        [
            "yt-dlg\\yt-dlg.exe",
            "yt-dlg"
        ]
    ],
    "checkver": {
        "github":"https://github.com/oleksis/youtube-dl-gui",
        "regex": "v(?<version>[\\d.]+)/(?<msi>.*).msi"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/$matchMsi.msi",
                "hash":{
                    "url": "$baseurl/SHA2-256SUMS"
                }
            }
        }
    }
}

```
Console logs:
```pwsh
❯ scoop install .\youtube-dl-gui.json
WARN  Purging previous failed installation of youtube-dl-gui.
Uninstalling 'youtube-dl-gui' ().
'youtube-dl-gui' was uninstalled.
Installing 'youtube-dl-gui' (1.8.3) [64bit]
Loading yt-dlg-20220118.2.msi from cache
Checking hash of yt-dlg-20220118.2.msi ... ok.
Extracting yt-dlg-20220118.2.msi ... done.
Linking ~\scoop\apps\youtube-dl-gui\current => ~\scoop\apps\youtube-dl-gui\1.8.3
Creating shortcut for yt-dlg (yt-dlg.exe)
'youtube-dl-gui' (1.8.3) was installed successfully!
```

**Comment by brian6932** [comment]
oh, I had to set `scoop config MSIEXTRACT_USE_LESSMSI false`, now it's fine, should that be included in `pre_isntall` somehow?

**Comment by brian6932** [comment]
So the 2 ways of getting this to work (for me) would be the following:
Direct .msi install:
```json
{
    "version": "1.8.3",
    "description": "A cross platform front-end GUI of the popular youtube-dl written in wxPython.",
    "homepage": "https://github.com/oleksis/youtube-dl-gui",
    "license": "Unlicense",
    "architecture": {
        "64bit": {
            "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v1.8.3/yt-dlg-20220118.2.msi",
            "hash": "71d4fc4eea97199218fdde36717e90326ed0fd4bd980c6afbffc263514e34be9"
        }
    },
    "shortcuts": [
        [
            "yt-dlg\\yt-dlg.exe",
            "yt-dlg"
        ]
    ],
    "checkver": {
        "github":"https://github.com/oleksis/youtube-dl-gui",
        "regex": "v(?<version>[\\d.]+)/(?<msi>.*).msi"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/$matchMsi.msi",
                "hash":{
                    "url": "$baseurl/SHA2-256SUMS"
                }
            }
        }
    }
}
```
with `#/dl.ism`:
```json
{
    "version": "1.8.3",
    "description": "A cross platform front-end GUI of the popular youtube-dl written in wxPython.",
    "homepage": "https://github.com/oleksis/youtube-dl-gui",
    "license": "Unlicense",
    "architecture": {
        "64bit": {
            "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v1.8.3/yt-dlg-20220118.2.msi#/dl.ism",
            "hash": "71d4fc4eea97199218fdde36717e90326ed0fd4bd980c6afbffc263514e34be9"
        }
    },
    "pre_install": [
        "scoop config MSIEXTRACT_USE_LESSMSI false",
        "$_args = @{",
        "    'Path' = \"$dir\\$fname\"",
        "    'DestinationPath' = \"$dir\"",
        "    'Removal' = $true",
        "}",
        "if (get_config 'MSIEXTRACT_USE_LESSMSI' $true) { $_args.Add('ExtractDir', 'APPDIR\\yt-dlg') } else { $_args.Add('ExtractDir', 'yt-dlg') }",
        "Expand-MsiArchive @_args"
    ],
    "shortcuts": [
        [
            "yt-dlg.exe",
            "yt-dlg"
        ]
    ],
    "checkver": {
        "github":"https://github.com/oleksis/youtube-dl-gui",
        "regex": "v(?<version>[\\d.]+)/(?<msi>.*).msi"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/$matchMsi.msi#/dl.ism",
                "hash":{
                    "url": "$baseurl/SHA2-256SUMS"
                }
            }
        }
    }
}

```
since people are having issues, is the latter preferable?

**Comment by brian6932** [comment]
Update: I swapped `scoop config MSIEXTRACT_USE_LESSMSI true` and it's still working without `"scoop config MSIEXTRACT_USE_LESSMSI false",`, I am not sure why my state was bugged, is there a reason why it could be between these 2 settings or something. It is a boolean, is it possible it was set to null somehow? I am very confused.

**Comment by HUMORCE** [comment]
`pre_install` script in this manifest (consider)for the MSI package is packaged in a special method.

```
PS C:\Users\HUMOR> scoop config MSIEXTRACT_USE_LESSMSI
True
PS C:\Users\HUMOR> scoop install 1ocal/youtube-dl-gui
Installing 'youtube-dl-gui' (1.8.3) [64bit]
Loading yt-dlg-20220118.2.msi from cache
Checking hash of yt-dlg-20220118.2.msi ... ok.
Running pre-install script...
Linking ~\scoop\apps\youtube-dl-gui\current => ~\scoop\apps\youtube-dl-gui\1.8.3
Creating shortcut for yt-dlg (yt-dlg.exe)
'youtube-dl-gui' (1.8.3) was installed successfully!
PS C:\Users\HUMOR> scoop uninstall youtube-dl-gui
Uninstalling 'youtube-dl-gui' (1.8.3).
Removing shortcut ~\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Scoop Apps\yt-dlg.lnk
Unlinking ~\scoop\apps\youtube-dl-gui\current
'youtube-dl-gui' was uninstalled.
PS C:\Users\HUMOR> scoop config MSIEXTRACT_USE_LESSMSI false
'MSIEXTRACT_USE_LESSMSI' has been set to 'false'
PS C:\Users\HUMOR> scoop install 1ocal/youtube-dl-gui
Installing 'youtube-dl-gui' (1.8.3) [64bit]
Loading yt-dlg-20220118.2.msi from cache
Checking hash of yt-dlg-20220118.2.msi ... ok.
Running pre-install script...
Linking ~\scoop\apps\youtube-dl-gui\current => ~\scoop\apps\youtube-dl-gui\1.8.3
Creating shortcut for yt-dlg (yt-dlg.exe)
'youtube-dl-gui' (1.8.3) was installed successfully!
```

File: bucket/youtube-dl-gui.json [hunk 3]
```diff
@@ -24,11 +24,17 @@
             "yt-dlg"
         ]
     ],
-    "checkver": "github",
+    "checkver": {
+        "github":"https://github.com/oleksis/youtube-dl-gui",
```
Comment (HUMORCE) [inline bucket/youtube-dl-gui.json:28]:
```suggestion
        "github": "https://github.com/oleksis/youtube-dl-gui",
```

File: bucket/youtube-dl-gui.json [hunk 4]
```diff
@@ -24,11 +24,17 @@
...
+        "regex": "v(?<version>[\\d.]+)/(?<msi>.*).msi"
+    },
     "autoupdate": {
         "architecture": {
             "64bit": {
-                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/yt-dlg.msi#/dl.ism"
+                "url": "https://github.com/oleksis/youtube-dl-gui/releases/download/v$version/$matchMsi.msi#/dl.ism",
+                "hash":{
```
Comment (HUMORCE) [inline bucket/youtube-dl-gui.json:35]:
```suggestion
                "hash": {
```

**Review by HUMORCE — CHANGES_REQUESTED** [review]
formatting

**Comment by brian6932** [comment]
> `pre_install` script in this manifest (consider)for the MSI package is packaged in a special method.

@HUMORCE I know, but if I had not previously defined the value `MSIEXTRACT_USE_LESSMSI`, what does it default to? I was getting errors previously, but after I set it to false, I stopped getting errors, even when I set it to true, I wasn't getting issues anymore. I am no longer able to reproduce my previous issues. The if else statement should cover if it's set to null, no? Is there some predefined value or null coalescing? I'm just trying to figure out what happened.

**Comment by HUMORCE** [comment]
you are right, i have always keep this option in the config, so i have not touch this error before.

```
PS C:\Users\HUMOR> scoop config rm MSIEXTRACT_USE_LESSMSI
'MSIEXTRACT_USE_LESSMSI' has been removed
PS C:\Users\HUMOR> scoop install 1ocal/youtube-dl-gui
Installing 'youtube-dl-gui' (1.8.3) [64bit]
Loading yt-dlg-20220118.2.msi from cache
Checking hash of yt-dlg-20220118.2.msi ... ok.
Running pre-install script...
DEBUG[1642914560] $out = -------------------------------------------------------------------------------
   ROBOCOPY     ::     Robust File Copy for Windows
-------------------------------------------------------------------------------

  Started : Sunday, January 23, 2022 5:09:20 AM
   Source : C:\Users\HUMOR\scoop\apps\youtube-dl-gui\1.8.3\_tmp\SourceDir\APPDIR\yt-dlg\
     Dest : C:\Users\HUMOR\scoop\apps\youtube-dl-gui\1.8.3\

    Files : *.*

  Options : *.* /S /E /DCOPY:DA /COPY:DAT /MOVE /R:1000000 /W:30

------------------------------------------------------------------------------

2022/01/23 05:09:20 ERROR 3 (0x00000003) Accessing Source Directory C:\Users\HUMOR\scoop\apps\youtube-dl-gui\1.8.3\_tmp\SourceDir\APPDIR\yt-dlg\
The system cannot find the path specified. -> C:\Users\HUMOR\scoop\apps\scoop\current\lib\core.ps1:515:9
Exception: C:\Users\HUMOR\scoop\apps\scoop\current\lib\core.ps1:516
Line |
 516 |          throw "Could not find '$(fname $from)'! (error $($proc.ExitCo …
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Could not find 'yt-dlg'! (error 16)
```

**Comment by HUMORCE** [comment]
test, then replace `pre_install` section with:

```
    "pre_install": [
        "$_args = @{",
        "    'Path' = \"$dir\\$fname\"",
        "    'DestinationPath' = \"$dir\"",
        "    'Removal' = $true",
        "}",
        "$LESSMSI = get_config 'MSIEXTRACT_USE_LESSMSI'",
        "if ($LESSMSI) { $_args.Add('ExtractDir', 'APPDIR\\yt-dlg') } else { $_args.Add('ExtractDir', 'yt-dlg') }",
        "Expand-MsiArchive @_args"
    ],
```

**Comment by brian6932** [comment]
@HUMORCE can confirm this fixed it, nice, you want to commit it, or should I?

**Comment by HUMORCE** [comment]
yes, pls

**Review by HUMORCE — APPROVED** [review]

**Comment by candrapersada** [comment]
why yt-dlg error for windows 7?
image
image
and why use `.msi` not `.exe`

**Comment by brian6932** [comment]
@candrapersada 
> why use `.msi` not `.exe`

Just preference, the `.msi` is more complete to me

> why yt-dlg error for windows 7?

You can't honestly expect people to support a 9 yr old OS

**Comment by candrapersada** [comment]
why is there no portable version like this?
```
		"if (!(Test-Path \"$dir\\User Data\\*\") -and (Test-Path \"$env:AppData\\yt-dlg\\User Data\")) {",
		"    info '[Portable Mode]: Copying user data...'",
		"    Copy-Item \"$env:AppData\\yt-dlg\\User Data\\*\" \"$dir\\User Data\" -Recurse",
		"}"
    ],
    "shortcuts": [
        [
            "yt-dlg.exe",
            "yt-dlg",
			"--user-data-dir=\"$dir\\User Data\""
```

**Comment by brian6932** [comment]
@candrapersada No one is stopping you from making one, just make a portable version, this has always used msi, even in the previous fork, so I just adjusted for the new one

**Pull request merged**